### PR TITLE
Issue/5461 media kill cursor

### LINKED ISSUE #5461: Drop the media browser cursor
The media browser relies on a SQlite cursor, which exposes the table schema outside of FluxC, is inefficient, and most importantly makes it hard to determine whether the dataset has actually changed since the last time the cursor was set.

Using `ArrayList<MediaModel>` would be a better approach.

### PR DESCRIPTION
**Pull request by nbradbury**
Fixes #5461 - restructures the media adapter to be based on an `ArrayList<MediaModel>` instead of a `Cursor`.

@aforcier Once this is merged, we can probably drop the cursor-based methods from the MediaStore.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/media/MediaGridAdapter.java [hunk 1]
```diff
@@ -463,23 +451,56 @@ public void run() {
...
+     * returns true if the passed list is the same as the existing one
+     */
+    private boolean isSameList(@NonNull List<MediaModel> mediaList) {
+        if (mediaList.size() != mMediaList.size()) {
+            return false;
+        }
+
+        for (MediaModel media: mediaList) {
+            // make sure this media item is in the existing media list
+            int index = indexOfId(media.getId());
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/media/MediaGridAdapter.java]: Isn't better to have a routine `getMediaWithID` that returns either the element or null. So we can just check for null, or check if the state are equals without 2 calls to `get` on `mMediaList`.

**Comment by daniloercoli** [comment]
This seems to fix all of the problems I saw on Media Library being out of synch when scrolling.

**Comment by aforcier** [comment]
> Once this is merged, we can probably drop the cursor-based methods from the MediaStore.

Unfortunately there's a bit more to do before we can drop them, we're still using cursors in [MediaGalleryEditFragment](https://github.com/wordpress-mobile/WordPress-Android/blob/cb401debe9e4aaa7be04f80e014c2babcc1516dc/WordPress/src/main/java/org/wordpress/android/ui/media/MediaGalleryEditFragment.java) + [MediaGalleryAdapter](https://github.com/wordpress-mobile/WordPress-Android/blob/cb401debe9e4aaa7be04f80e014c2babcc1516dc/WordPress/src/main/java/org/wordpress/android/ui/media/MediaGalleryAdapter.java).

**Comment by nbradbury** [comment]
> Isn't better to have a routine getMediaWithID that returns either the element or null.

Good call, I agree. Added in c6c4618

**Comment by nbradbury** [comment]
@daniloercoli Ready for another round!

**Comment by daniloercoli** [comment]
:shipit:

**Pull request merged**